### PR TITLE
Migrate Payment methods page

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrates 'Improve > Payment > Payment Methods' page to Symfony
| Type?         | new feature
| Category?     | BO
| BC breaks?    | yes (overrides)
| Deprecations? |no
| Fixed ticket? | n/a
| How to test?  | New page should work the same as old one.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9192)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    /**
+     * Gets installed and enabled payment modules for PaymentMethods controller to display
+     *
+     * @param string $legacyController
+     *
+     * @return ModuleInterface[]
+     */
+    protected function getPaymentModulesToDisplay($legacyController)
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php]: to be extracted in its own service :+1:

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          </a>
+        </div>
+      </div>
+    </div>
+  {% endfor %}
+  </div>
+  {% else %}
+    <div class="alert alert-warning" role="alert">
+      <p class="alert-text">
+        {{ 'It seems there are no recommended payment solutions for your country.'|trans }} <br>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: Is this an error message?

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        <a class="_blank" href="https://www.prestashop.com/en/contact-us">{{ 'Do you think there should be one? Let us know!'|trans }}</a>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: It is currently in both Admin.Shipping.Feature and messages files.

Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: should it keep `Admin.Shipping.Feature` domain?

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig (hunk 2 above)
Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: It's more like notification/warning. But not an error.

File: src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php (hunk 1 above)
Comment (sarjon) [reply src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php]: any suggestions were it should go? :thinking:

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig (hunk 2 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: If it's not a tooltip, I would say _Admin.Payment.Notification_ then!

File: src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig (hunk 3 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/PaymentMethods/Blocks/payment_modules_list.html.twig]: I think so... what do you think @Quetzacoalt91 ?

File: src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php (hunk 1 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentMethodsController.php]: in ``PrestaShop\PrestaShop\Adapter\Module\Presenter\PaymentModulesPresenter`` with only one method: ``present()`` with no arguments, you can declare "legacyController" as constant for exemple.

**Comment by mickaelandrieu** [comment]
Now we have the new routing, you need to update your pull request 👍 

I'll document this asap, thanks @sarjon !

**Comment by sarjon** [comment]
@mickaelandrieu rebased and ready for review/QA!

**Review by mickaelandrieu — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]
👍

**Comment by mickaelandrieu** [comment]
Thanks @sarjon !

**Pull request merged**